dlang/source/flatbuffers/table.d

```diff
@@ -0,0 +1,210 @@
+    }
+
+    /// Retrieve the relative offset stored at "offset".
+	uint __indirect(size_t offset)
+    {
+		return cast(uint)(offset + _buffer.get!int(offset));
+    }
+
+    /// Create a D string from UTF-8 data stored inside the flatbuffer.
+	string __string(size_t offset)
+    {
+        offset += _buffer.get!int(offset);
+        auto len = _buffer.get!uint(offset);
+        auto startPos = offset + uint.sizeof;
+        return cast(string) _buffer.data[startPos .. startPos + len];
+    }
+
+    /// Get the length of a vector whose offset is stored at "offset" in this object.
+	uint __vector_len(size_t offset)
+    {
+        offset += _pos;
+        offset += _buffer.get!int(offset);
+        return _buffer.get!uint(offset);
+    }
+
+    /// Get the start of data of a vector whose offset is stored at "offset" in this object.
+	uint __dvector(size_t offset)
+    {
+        offset += _pos;
+		return cast(uint)(offset + _buffer.get!int(offset) + int.sizeof); // Data starts after the length.
+    }
+
+    /// Initialize any Table-derived type to point to the union at the given offset.
+	T __union(T)(size_t offset)
+    {
+        offset += _pos;
+        return T.init_((offset + _buffer.get!int(offset)), _buffer);
+    }
+
+    static bool __has_identifier(ByteBuffer bb, string ident)
+    {
+        import std.string;
+
+        if (ident.length != fileIdentifierLength)
+            throw new ArgumentException(
+                format("FlatBuffers: file identifier must be length %s.", fileIdentifierLength),
+                "ident");
+
+        for (auto i = 0; i < fileIdentifierLength; i++)
+        {
+            if (ident[i] != cast(char) bb.get!byte(bb.position() + cast(uint)uint.sizeof + i))
+                return false;
+        }
+
+        return true;
+    }
+}
+
+import std.traits;
+
+/**
+    Iterator for the vector.
+*/
+struct Iterator(ParentType, ReturnType, string accessor)
+{
+    static if (isScalarType!(ReturnType) || isSomeString!(ReturnType))
+        alias ApplyType = ReturnType;
+    else
+        alias ApplyType = Nullable!ReturnType;
+private:
+        ParentType parent;
+    int index;
+public:
+    this(ParentType parent)
+    {
+        this.parent = parent;
+    }
+
+    @property int length()
+    {
+        mixin("return parent." ~ accessor ~ "Length;");
+    }
+
+    @property bool empty()
+    {
+        return index == length;
+    }
+
+    @property ReturnType popFront()
+    {
+        mixin("return parent." ~ accessor ~ "(++index);");
+    }
+
+    @property ReturnType front()
+    {
+        mixin("return parent." ~ accessor ~ "(index);");
+    }
+
+    ReturnType opIndex(int index)
+    {
+        mixin("return parent." ~ accessor ~ "(index);");
+    }
+
+    int opApply(int delegate(ApplyType) operations)
+    {
+        int result = 0;
+        for (int number = 0; number < length(); ++number)
+        {
+            static if (isScalarType!(ReturnType) || isSomeString!(ReturnType))
```

Comment:
`static if (isScalarType!ReturnType || isSomeString!ReturnType)`  maybe? (no template invocation parentheses for single argument invocation.